samples/quantum-approximation-optimization/qaoa-for-quadratic-unconstrained-binary-optimization.ipynb

```diff
@@ -0,0 +1,801 @@
+    "        name=\"Pasqal sequence\",\n",
+    "        input_params={\n",
+    "            \"shots\": 10\n",
+    "        }\n",
+    "    )\n",
+    "\n",
+    "    print(f\"Queued job: {job.id}\")\n",
+    "    job.wait_until_completed()\n",
+    "    print(f\"Job completed with state: {job.details.status}\")\n",
+    "    result = job.get_results()\n",
+    "\n",
+    "    return result"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Please set your `resource_id` and `location` below."
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "from azure.quantum import Workspace\n",
+    "workspace = Workspace(\n",
+    "            resource_id = \"\",\n",
+    "            location = \"\",\n",
+    ")\n",
+    "target = workspace.get_targets(name=\"pasqal.sim.emu-tn\")\n",
+    "\n",
+    "if isinstance(target, list):\n",
+    "    if len(target) == 0:\n",
+    "        raise RuntimeError(\"No emulators found for that name\")\n",
+    "    raise RuntimeError(target)\n",
+    "\n",
+    "# Open a session on the backend to group all jobs together\n",
+    "from azure.quantum.job.session import Session, SessionJobFailurePolicy \n",
+    "session = target.open_session(name=\"QUBO Problem\", job_failure_policy=SessionJobFailurePolicy.CONTINUE)\n",
+    "target.latest_session = session"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": "Note: the next cell may take up to 5 minutes to complete."
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "# Compute the time evolution on PASQAL emulator\n",
+    "cost_emutn = []\n",
+    "try:\n",
+    "    for T in 1000 * np.linspace(1, 10, 10):\n",
+    "        seq = Sequence(reg, DigitalAnalogDevice)\n",
+    "        seq.declare_channel(\"ising\", \"rydberg_global\")\n",
+    "        adiabatic_pulse = Pulse(\n",
+    "            InterpolatedWaveform(T, [1e-9, Omega, 1e-9]),\n",
+    "            InterpolatedWaveform(T, [delta_0, 0, delta_f]),\n",
+    "            0,\n",
+    "        )\n",
+    "        seq.add(adiabatic_pulse, \"ising\")\n",
+    "\n",
+    "        from collections import Counter\n",
+    "        histogram_results = submit_job(target, seq)\n",
+    "        count_dict = Counter(histogram_results)\n",
+    "\n",
+    "        cost_emutn.append(get_cost(count_dict, Q) / 3)\n",
+    "\n",
+    "except Exception as e:\n",
+    "    print(e)\n",
+    "session.close()"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "code",
+   "metadata": {},
+   "source": [
+    "plot_cost(cost_emutn, \"Pasqal Tensor Network-based emulator\")"
+   ],
+   "outputs": [],
+   "execution_count": null
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": "#### Finally, let's compare the results of the two emulators"
```

Comment:
Is there an opportunity for us to add a summery at the end and explain the difference of results produced by `QutipEmulator` and PASQAL emulator?
For instance:
- Why are they different
- Which emulator has higher fidelity and why
- Which emulator should be used in which case (trade-offs like `QutipEmulator` is quicker to produce the result, but is less accurate, `QutipEmulator` doesn't converge is certain cases therefore couldn't be used, etc)

Reply:
I would need to loop back internally to answer these questions, and I'm a bit worried about how long that could take.

Would it be possible to merge this PR, and refine this point in a subsequent PR ?

Reply:
Sounds good! Let's merge it and do a follow up PR later. Sorry for so much churn on this PR and thank you for working with me on improving it!

Reply:
Thank you very much :relieved:

I have requested this to be prioritized on our side.